robocop/checkers/misc.py

```diff
@@ -339,6 +348,27 @@
             END
         """,
     ),
+    "0919": Rule(
+        rule_id="0919",
+        name="unused-argument",
+        msg="Keyword '{{ arg }}' argument is not used",
+        severity=RuleSeverity.WARNING,
+        docs="""
+        Keyword argument was defined but not used::
+        
+            *** Keywords ***
+            Keyword
+                [Arguments]    ${used}    ${overwritten_before_used}    ${not_used}
+                Log    ${used}
+                ${overwritten_before_used}    Process Stuff    value
+                Log    ${overwritten_before_used}  # used, but original argument was overwritten before using
```

Comment:
I'm wondering if it's good we detect such cases in the `unused-argument` rule, since this is a bit different case and the users might actually not expect it to work that way. Shouldn't we split this into 2 separate rules: `unused-argument` and maybe `overwritten-before-usage`?

Reply:
I was actually thinking that this rule would be good base for other rules, such as ``unused-local-variable`` rule. It's also similar and it could be also handled by this checker (since we're already doing most of the heavy lifting with parsing the variables etc). And it would be more more obvious to the users.

tl;dr; I agree, let's split it into ``overwritten-before-usage`` rule. In that case I should probably do this in the same PR. I will create issue for it (let's discuss naming of the rule and misc stuff here), I will link it

Reply:
#836

Reply:
Maybe, to make it more explicit, we should name it `overwritten-arg-before-usage`? Or even `overwritten-argument-before-usage`? It makes it a bit longer, but it better points out what is this rule about. WDYT? Or maybe just `overwritten-argument`?

Reply:
Hmm... Yeah, right. Well, does it have any negative impact if we create another separate rule for overwritten local variable? I might be missing something, but I think higher granularity is fine.

Reply:
Yeah, Im also for more granulity as long it makes sense. Makes it easier to understand issue /disable rule if needed. I will do it then